src/library/scala/collection/mutable/HashMap.scala

```diff
@@ -161,7 +161,13 @@ class HashMap[K, V](initialCapacity: Int, loadFactor: Double)
           else table(indexedHash) = foundNode.next
           contentSize -= 1
 
-        case (None, Some(value)) => put0(key, value, false, hash, indexedHash)
+        case (None, Some(value)) =>
+          val newIndexedHash =
+            if (contentSize + 1 >= threshold) {
+              growTable(table.length * 2)
```

Comment:
not really specific to this PR, but the `growTable` method is broken (does not handle integer overflow, computes a `threshold` value but never uses it)

Reply:
That might be true I haven't gotten that deep but performance tests are much better with this change and the table expands so, as you said, this should probably be a separate issue.